plugins/in_dummy/in_dummy.c

```diff
@@ -32,29 +33,173 @@
 
 #include "in_dummy.h"
 
-
 static int set_dummy_timestamp(msgpack_packer *mp_pck, struct flb_dummy *ctx)
 {
     struct flb_time t;
     struct flb_time diff;
     struct flb_time dummy_time;
     int ret;
 
-    if (ctx->base_timestamp == NULL) {
-        ctx->base_timestamp = flb_malloc(sizeof(struct flb_time));
-        flb_time_get(ctx->base_timestamp);
-        ret = flb_time_append_to_msgpack(ctx->dummy_timestamp, mp_pck, 0);
+    if (ctx->data.log.base_timestamp == NULL) {
+        ctx->data.log.base_timestamp = flb_malloc(sizeof(struct flb_time));
+        if (!ctx->data.log.base_timestamp) {
+            flb_errno();
+            return -1;
+        }
+
+        flb_time_get(ctx->data.log.base_timestamp);
+        ret = flb_time_append_to_msgpack(ctx->data.log.dummy_timestamp, mp_pck, 0);
     }
     else {
         flb_time_get(&t);
-        flb_time_diff(&t, ctx->base_timestamp, &diff);
-        flb_time_add(ctx->dummy_timestamp, &diff, &dummy_time);
+        flb_time_diff(&t, ctx->data.log.base_timestamp, &diff);
+        flb_time_add(ctx->data.log.dummy_timestamp, &diff, &dummy_time);
         ret = flb_time_append_to_msgpack(&dummy_time, mp_pck, 0);
     }
 
     return ret;
 }
 
+static int gen_trace(struct flb_input_instance *ins, void *in_context)
```

Comment:
Does the function create static ctrace data ?
Is it necessary to make it configurable ?

If so, I think it may be better to create another plugin.
Because configuration parameters will be different from parameters for "log".
Data sets are also different.  It may be complex for user and developer.
```c
    union {
        struct log_message log;
        struct ctrace *trace;
    } data;
```